docs/vision-usage.rst

```diff
@@ -0,0 +1,190 @@
+Using the Vision API
+====================
+
+Authentication and Configuration
+--------------------------------
+
+- For an overview of authentication in ``gcloud-python``,
+  see :doc:`gcloud-auth`.
+
+- In addition to any authentication configuration, you should also set the
+  :envvar:`GCLOUD_PROJECT` environment variable for the project you'd like
+  to interact with. If you are Google App Engine or Google Compute Engine
+  this will be detected automatically.
+
+- After configuring your environment, create a
+  :class:`Client <gcloud.vision.client.Client>`
+
+  .. doctest::
+
+     >>> from gcloud import vision
+     >>> client = vision.Client()
+
+  or pass in ``credentials`` and ``project`` explicitly
+
+  .. doctest::
+
+     >>> from gcloud import vision
+     >>> client = vision.Client(project='my-project', credentials=creds)
+
+Annotating an Image
+-------------------
+
+Annotate a single image
+~~~~~~~~~~~~~~~~~~~~~~~
+
+.. doctest::
+
+    >>> from gcloud import vision
+    >>> client = vision.Client(project="my-project-name")
+    >>> with open('/tmp/car.jpg', 'r') as f:
+    ...     client.annotate(f.read(), vision.FeatureTypes.LABEL_DETECTION, 3)
```

Comment:
@jonparrott , for batching, wouldn't we just throw our methods into a context manager that reconciles all this stuff?

For example, here's what I envision as the code I'd write to do a bunch of detecting:

```python
image1 = client.image('/path/to/image1.jpg')
image2 = client.image('/path/to/image2.jpg')

faces = image1.detect_faces(limit=10)
labels = image1.detect_labels()
other_image_has_face = image2.has_faces()

# Now you have all the stuff you want, but you made 3 different API calls for it.
```

You then realize that this is too slow, and you want to batch this up into a single API call...

```python
image1 = client.image('/path/to/image1.jpg')
image2 = client.image('/path/to/image2.jpg')

with client.batch():
  faces = image1.detect_faces(limit=10)
  labels = image1.detect_labels()
  other_image_has_face = image2.has_faces()

# Now you have all the stuff you want, and we bulked it into a single request.
```